.github/workflows/ci.yml

```diff
@@ -0,0 +1,237 @@
+      - run: pip install ".[test]" pytest-xdist
+      - run: pip freeze
+      - run: pytest -n auto
+  test_alldeps:
+    name: test optional dependencies
+    needs: [ test ]
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ ubuntu-latest, macos-latest ]
+        python: [ 3.8, 3.9, '3.10' ]
+    steps:
+      - uses: actions/checkout@v3.0.2
+      - uses: actions/setup-python@v4.1.0
+        id: python
+        with:
+          python-version: ${{ matrix.python }}
+      - uses: actions/cache@v3.0.5
+        with:
+          path: ${{ env.pythonLocation }}
+          key: test-alldeps-${{ runner.os }}-${{ env.pythonLocation }}-${{ hashFiles('**/pyproject.toml', '**/setup.*') }}
+      - run: pip install ".[test,all]" pytest-xdist
+      - run: pip freeze
+      - run: pytest -n auto
+  test_devdeps:
+    name: test developer versions
+    needs: [ test ]
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ ubuntu-latest, macos-latest ]
+        python: [ 3.8, 3.9, '3.10' ]
+    steps:
+      - uses: actions/checkout@v3.0.2
+      - uses: actions/setup-python@v4.1.0
+        id: python
+        with:
+          python-version: ${{ matrix.python }}
+      - uses: actions/cache@v3.0.5
+        with:
+          path: ${{ env.pythonLocation }}
+          key: test-devdeps-${{ runner.os }}-${{ env.pythonLocation }}-${{ hashFiles('**/pyproject.toml', '**/setup.*') }}
+      - run: pip install git+https://github.com/asdf-format/asdf git+https://github.com/spacetelescope/stpipe git+https://github.com/spacetelescope/stdatamodels --pre astropy numpy
+      - run: pip install ".[test]" pytest-xdist
+      - run: pip freeze
+      - run: pytest -n auto
+  test_pyargs:
+    name: test --pyargs
+    needs: [ test ]
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ ubuntu-latest, macos-latest ]
+        python: [ 3.8, 3.9, '3.10' ]
+        exclude:
+          - os: macos-latest
+            python: '3.10'
+    steps:
+      - uses: actions/checkout@v3.0.2
+      - uses: actions/setup-python@v4.1.0
+        id: python
+        with:
+          python-version: ${{ matrix.python }}
+      - uses: actions/cache@v3.0.5
+        with:
+          path: ${{ env.pythonLocation }}
+          key: test-pyargs-${{ runner.os }}-${{ env.pythonLocation }}-${{ hashFiles('**/pyproject.toml', '**/setup.*') }}
+      - run: pip install ".[test]" pytest-xdist
+      - run: pip freeze
+      - run: pytest -n auto ./docs --pyargs romanisim
+  test_older_numpy:
+    name: test Numpy ${{ matrix.numpy }} (Python ${{ matrix.python }})
+    needs: [ test ]
+    runs-on: ${{ matrix.os }}
+    continue-on-error: true
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ ubuntu-latest ]
+        python: [ 3.8, 3.9, '3.10' ]
+        numpy: [ '1.18.*', '1.19.*', '1.20.*', '1.21.*', '1.22.*' ]
+        exclude:
```

Comment:
I'd consider dropping support numpy 3.18 and maybe 3.19?

Reply:
Astropy is dropping support in the next release for numpy 1.18 and 1.19, so we should follow the same path here.

Reply:
Removed.